ruma-events/src/event_kinds.rs

```diff
@@ -149,6 +191,76 @@ pub struct StrippedStateEventStub<C: StateEventContent> {
     pub state_key: String,
 }
 
+/// A redacted state event.
+#[derive(Clone, Debug, Event)]
+pub struct RedactedStateEvent<C: RedactedStateEventContent> {
+    /// Data specific to the event type.
+    pub content: C,
+
+    /// The globally unique event identifier for the user who sent the event.
+    pub event_id: EventId,
+
+    /// The fully-qualified ID of the user who sent this event.
+    pub sender: UserId,
+
+    /// Timestamp in milliseconds on originating homeserver when this event was sent.
+    pub origin_server_ts: SystemTime,
+
+    /// The ID of the room associated with this event.
+    pub room_id: RoomId,
+
+    /// A unique key which defines the overwriting semantics for this piece of room state.
+    ///
+    /// This is often an empty string, but some events send a `UserId` to show
+    /// which user the event affects.
+    pub state_key: String,
+
+    /// Additional key-value pairs not signed by the homeserver.
+    pub unsigned: UnsignedData,
+}
+
+/// A redacted state event without a `room_id`.
+#[derive(Clone, Debug, Event)]
+pub struct RedactedStateEventStub<C: RedactedStateEventContent> {
+    /// Data specific to the event type.
+    // #[serde(default, skip_serializing_if = "is_zst")]
+    pub content: C,
+
+    /// The globally unique event identifier for the user who sent the event.
+    pub event_id: EventId,
+
+    /// The fully-qualified ID of the user who sent this event.
+    pub sender: UserId,
+
+    /// Timestamp in milliseconds on originating homeserver when this event was sent.
+    pub origin_server_ts: SystemTime,
+
+    /// A unique key which defines the overwriting semantics for this piece of room state.
+    ///
+    /// This is often an empty string, but some events send a `UserId` to show
+    /// which user the event affects.
+    pub state_key: String,
+
+    /// Additional key-value pairs not signed by the homeserver.
+    pub unsigned: UnsignedData,
+}
+
+/// A stripped-down redacted state event.
+#[derive(Clone, Debug, Event)]
+pub struct RedactedStrippedStateEventStub<C: RedactedStateEventContent> {
```

Comment:
No `RedactedStrippedStateEvent`?

Reply:
Is there supposed to also be `StrippedStateEvent`? If so there is no regular event struct for this either.

Reply:
Oh, right. 😄 

Yeah no, stripped state events are only ever received as part of the sync response ^^

Reply:
So I don't need to change anything right?

Reply:
No ^^